walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,48 +13,84 @@
 // limitations under the License.
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
 
+
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> availableTimes = new ArrayList<TimeRange>();
-    List<TimeRange> unavailableTimes = new ArrayList<TimeRange>();
-
+    List<TimeRange> availableTimesForMandatory = new ArrayList<TimeRange>();
+    List<TimeRange> availableTimesForOptional = new ArrayList<TimeRange>();
+    List<TimeRange> unavailableTimesForMandatory = new ArrayList<TimeRange>();
+    List<TimeRange> unavailableTimesForOptional = new ArrayList<TimeRange>();
+    
     for (Event e : events) {
-        if (!attendeesFree(e, request)) {
-            unavailableTimes.add(e.getWhen());
+        if (!mandatoryAttendeesFree(e, request)) {
+            unavailableTimesForMandatory.add(e.getWhen());
+        }
+
+        if (!optionalAttendeesFree(e, request)) {
+            unavailableTimesForOptional.add(e.getWhen());
         }
     }
 
-    Collections.sort(unavailableTimes, TimeRange.ORDER_BY_START);
+    Collections.sort(unavailableTimesForMandatory, TimeRange.ORDER_BY_START);
+    Collections.sort(unavailableTimesForOptional, TimeRange.ORDER_BY_START);
 
-    if (unavailableTimes.size() > 0) {
-        combineOverlaps(unavailableTimes, unavailableTimes.get(0), 0);
+    if (unavailableTimesForMandatory.size() > 0) { // Do this only for mandatory attendees since only one optional attendee exists
```

Comment:
Looks like this exceeds the recommended column limit: 
https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

Reply:
Oh I did not know that there was a column limit, thank you for the information. Also, thank you for approving the PR :)